src/deepagents/middleware/filesystem.py

```diff
@@ -0,0 +1,1109 @@
+
+        Raises:
+            ValueError: If long_term_memory is True but runtime.store is None.
+        """
+        if self.long_term_memory and runtime.store is None:
+            msg = "Longterm memory is enabled, but no store is available"
+            raise ValueError(msg)
+        return request
+
+    def wrap_model_call(
+        self,
+        request: ModelRequest,
+        handler: Callable[[ModelRequest], ModelResponse],
+    ) -> ModelResponse:
+        """Update the system prompt to include instructions on using the filesystem.
+
+        Args:
+            request: The model request being processed.
+            handler: The handler function to call with the modified request.
+
+        Returns:
+            The model response from the handler.
+        """
+        if self.system_prompt is not None:
+            request.system_prompt = request.system_prompt + "\n\n" + self.system_prompt if request.system_prompt else self.system_prompt
+        return handler(request)
+
+    async def awrap_model_call(
+        self,
+        request: ModelRequest,
+        handler: Callable[[ModelRequest], Awaitable[ModelResponse]],
+    ) -> ModelResponse:
+        """(async) Update the system prompt to include instructions on using the filesystem.
+
+        Args:
+            request: The model request being processed.
+            handler: The handler function to call with the modified request.
+
+        Returns:
+            The model response from the handler.
+        """
+        if self.system_prompt is not None:
+            request.system_prompt = request.system_prompt + "\n\n" + self.system_prompt if request.system_prompt else self.system_prompt
+        return await handler(request)
+
+    def wrap_tool_call(
+        self,
+        request: ToolCallRequest,
+        handler: Callable[[ToolCallRequest], ToolMessage | Command],
+    ) -> ToolMessage | Command:
+        """Check the size of the tool call result and evict to filesystem if too large.
+
+        Args:
+            request: The tool call request being processed.
+            handler: The handler function to call with the modified request.
+
+        Returns:
+            The raw ToolMessage, or a pseudo tool message with the ToolResult in state.
+        """
+        # If no token limit specified, or if it is a filesystem tool, do not evict
+        if self.tool_token_limit_before_evict is None or request.tool_call["name"] in TOOL_GENERATORS:
+            return handler(request)
+
+        tool_result = handler(request)
+
+        # Only handle ToolMessages, not Commands
+        if isinstance(tool_result, ToolMessage) and isinstance(tool_result.content, str):
+            content = tool_result.content
+            # Estimating tokens to be 4 characters per token
+            if len(content) > 4 * self.tool_token_limit_before_evict:
+                file_path = f"/large_tool_results/{tool_result.tool_call_id}"
+                tool_large_tool_msg = (
```

Comment:
id include the first 10 lines (each line capped at 1000 chars or something) in this to give it some sense